pkg/reconciler/secret/reconciler.go

```diff
@@ -24,83 +24,149 @@ import (
 	"github.com/pkg/errors"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
+	"github.com/crossplaneio/crossplane-runtime/pkg/event"
 	"github.com/crossplaneio/crossplane-runtime/pkg/logging"
 	"github.com/crossplaneio/crossplane-runtime/pkg/resource"
 )
 
-const (
-	secretControllerName   = "secretpropagator.crossplane.io"
-	secretReconcileTimeout = 1 * time.Minute
+const secretReconcileTimeout = 1 * time.Minute
 
+// Error messages.
+const (
 	errGetSecret         = "cannot get managed resource's connection secret"
 	errUpdateSecret      = "cannot update connection secret"
 	errUnexpectedFromUID = "unexpected propagate from uid on propagated secret"
 	errUnexpectedToUID   = "unexpected propagate to uid on propagator secret"
 )
 
-var log = logging.Logger.WithName("controller")
+// Event reasons
+const (
+	reasonPropagatedFrom event.Reason = "PropagatedDataFrom"
+	reasonPropagatedTo   event.Reason = "PropagatedDataTo"
+)
+
+// ControllerName returns the recommended name for controllers that use this
+// package to reconcile a particular kind of resource claim.
+func ControllerName(kind string) string {
+	return "secretpropagating/" + strings.ToLower(kind)
+}
+
+// Reconciler reconciles secrets by propagating their data from another secret.
+// Both secrets must consent to this process by including propagation
+// annotations. The Reconciler assumes it has a watch on both propagating (from)
+// and propagated (to) secrets.
+type Reconciler struct {
```

Comment:
This will be a nice help with the making the `Secret` reconciler implementation look more like the others

Reply:
Yeah, it was essentially required now that we want functional arguments.